packet.go

```diff
@@ -709,12 +727,25 @@ func (p *sshFxpOpenPacket) UnmarshalBinary(b []byte) error {
 		return err
 	} else if p.Pflags, b, err = unmarshalUint32Safe(b); err != nil {
 		return err
-	} else if p.Flags, _, err = unmarshalUint32Safe(b); err != nil {
+	} else if p.Flags, b, err = unmarshalUint32Safe(b); err != nil {
 		return err
 	}
+	p.Attrs = b
 	return nil
 }
 
+func (p *sshFxpOpenPacket) unmarshalFileStat(flags uint32) *FileStat {
+	switch attrs := p.Attrs.(type) {
+	case *FileStat:
+		return attrs
+	case []byte:
+		fs, _ := unmarshalFileStat(flags, attrs)
+		return fs
+	default:
+		panic(fmt.Sprintf("invalid type in unmarshalFileStat: %T", attrs))
```

Comment:
“don’t panic” but also… we don’t have an `error`?

All the more reason why `unmarshalFileStat` should return an `error`.